### ISSUE TITLE
3D Graph Visualizations

### ISSUE THREAD

**Issue by Erstwild** [bokeh/bokeh#3693]
I had heard some recent discussion that perhaps a limited subset of 3D rendering options might be introduced in Bokeh sometime this year.  I did want to inquire if perhaps 3D graph visualizations might be targeted (ala, gephi)?

**Comment by catubc** [comment]
Any comment on this? I'm really interested in this also... am trying to avoid re-learning vispy )

**Comment by datnamer** [comment]
Maybe vispy can be used as a backend for a custom bokeh model

**Comment by catubc** [comment]
What exactly would that involve?  I work very far from web apps... but might be willing if straight forward path..

**Comment by bryevdv** [comment]
There's currently a branch with some work, it would look roughly like this: https://github.com/bokeh/bokeh/tree/surface3d_custom

However, that branch hardcodes the necessary 3rd party 3d JS library. What is really needed first, to be able to do this, is a way to specify external JS libs to load as resources. There is an issue open for tackling that: https://github.com/bokeh/bokeh/issues/4127 Once that is complete we will definitely add some examples of wrapping some external JS libraries, including 3d vis ones.

**Comment by ijstokes** [comment]
Any idea what the timeline is for getting some of this March branch into an upcoming release?

**Comment by birdsarah** [comment]
I don't think this would ever be in core bokeh. 

Maybe an add-ons package like @brendancol is making for gis stuff.

**Comment by bryevdv** [comment]
The 3d user *custom model example* should be in the upcoming `0.12` release. I agree that it's probably out of scope for the core library. `0.12` is a bit behind schedule, but it should be ready in the next few weeks. We are looking at workflows to support frequent, small, time-based bugfix releases starting in the near future.

**Comment by bryevdv** [comment]
@Erstwild @datnamer @ijstokes @catubc please check out https://github.com/bokeh/bokeh/pull/4592

This is just a quick and dirt demo, I'd like to wrap a better 3D JS library more completely for a better example, but perhaps this will light a path.
